### PR TITLE
KCHd Integration

### PR DESCRIPTION
**Pull request by trickeydan**

### CODE REVIEW

File: sr/robot3/robot.py [hunk 1]
```diff
@@ -12,8 +12,8 @@
 from j5 import __version__ as j5_version
 from j5.backends import Backend
 from j5.boards import Board, BoardGroup
-from j5.boards.sr.v4 import MotorBoard, PowerBoard, ServoBoard
-from j5.boards.sr.v4.ruggeduino import Ruggeduino
+from j5.boards.sr import KCHBoard
+from j5.boards.sr.v4 import MotorBoard, PowerBoard, Ruggeduino, ServoBoard
```
Comment (WillB97) [inline sr/robot3/robot.py:16]: Has the ruggeduino actually moved?

Comment (trickeydan) [reply sr/robot3/robot.py:16]: No, the import should always have been like this.

File: sr/robot3/kch.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        self._mqtt.subscribe("kchd", self._handle_kchd_message)
+
+    @property
+    def name(self) -> str:
+        """
+        MQTT client name of the data component.
+
+        This should be unique, as clashes will cause unexpected disconnections.
+        """
+        return f"sr-robot3-kchd-{uuid4()}"
```
Comment (PeterJCLaw) [inline sr/robot3/kch.py]: This is going to return a different value each time it's accessed. Is that expected?
Perhaps this should be a [`cached_property`](https://docs.python.org/3/library/functools.html#functools.cached_property)?

This also applies to `KCHDaemonControlConsumer.name` below.

Comment (trickeydan) [reply sr/robot3/kch.py]: I think you are correct. This issue probably also applies to code in `astoria.py` and `astctl`.

Comment (raccube) [reply sr/robot3/kch.py]: 8e16685

File: sr/robot3/kch.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        self._kchdc = KCHDaemonControlConsumer(False, None)
+        try:
+            asyncio.ensure_future(self._kchdc.run())
+        except ConnectionRefusedError:
+            LOGGER.warning("Unable to connect to MQTT broker")
+        except asyncio.TimeoutError:
+            LOGGER.warning("KCH Daemon took too long to respond, giving up.")
+
+    async def _set_leds(self) -> None:
+        await self._kchdc._ready.wait()
```
Comment (WillB97) [inline sr/robot3/kch.py]: We could add a timeout on the wait so it doesn't hang indenfinitely if kchd crashes.

```suggestion
    try:
        await asyncio.wait_for(self._kchdc._ready.wait(), timeout=1.0)
    except asyncio.TimeoutError:
        LOGGER.error("Failed to set LED on KCH")
        return
```

**Review by WillB97 — APPROVED** [review]
It'll work. We could add some extra protection to reduce the chance of it getting stuck if something else breaks.

**Pull request merged**